src/material-date-fns-adapter/adapter/date-fns-adapter.ts

```diff
@@ -241,4 +246,42 @@ export class DateFnsAdapter extends DateAdapter<Date, Locale> {
   invalid(): Date {
     return new Date(NaN);
   }
+
+  override setTime(target: Date, hours: number, minutes: number, seconds: number): Date {
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      if (hours < 0 || hours > 23) {
+        throw Error(`Invalid hours "${hours}". Hours value must be between 0 and 23.`);
+      }
+
+      if (minutes < 0 || minutes > 59) {
+        throw Error(`Invalid minutes "${minutes}". Minutes value must be between 0 and 59.`);
+      }
+
+      if (seconds < 0 || seconds > 59) {
+        throw Error(`Invalid seconds "${seconds}". Seconds value must be between 0 and 59.`);
+      }
+    }
+
+    return set(this.clone(target), {hours, minutes, seconds});
```

Comment:
should we also be normalizing by zero-ing out the millis?

Reply:
Yeah good idea. Changed.